experimental/traffic-portal/src/app/core/origins/detail/origin-detail.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+	let component: OriginDetailComponent;
+	let fixture: ComponentFixture<OriginDetailComponent>;
+	let route: ActivatedRoute;
+	let paramMap: jasmine.Spy;
+	let loader: HarnessLoader;
+	let service: OriginService;
+
+	const navSvc = jasmine.createSpyObj([], {
+		headerHidden: new ReplaySubject<boolean>(),
+		headerTitle: new ReplaySubject<string>(),
+	});
+	beforeEach(async () => {
+		await TestBed.configureTestingModule({
+			declarations: [OriginDetailComponent],
+			imports: [APITestingModule, RouterTestingModule, MatDialogModule],
+			providers: [{provide: NavigationService, useValue: navSvc}],
+		}).compileComponents();
+
+		route = TestBed.inject(ActivatedRoute);
+		paramMap = spyOn(route.snapshot.paramMap, "get");
+		paramMap.and.returnValue(null);
+		fixture = TestBed.createComponent(OriginDetailComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+		service = TestBed.inject(OriginService);
+		loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+		await fixture.whenStable();
+	});
+
+	it("should create", () => {
+		expect(component).toBeTruthy();
+	});
+
+	it("new origin", async () => {
+		paramMap.and.returnValue("new");
+
+		fixture = TestBed.createComponent(OriginDetailComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+		await fixture.whenStable();
+		expect(paramMap).toHaveBeenCalled();
+		expect(component.origin).not.toBeNull();
+		expect(component.origin.name).toBe("");
+		expect(component.new).toBeTrue();
+	});
+
+	it("existing origin", async () => {
+		const id = 1;
+		paramMap.and.returnValue(id);
+		const origin = await service.getOrigins(id);
+		fixture = TestBed.createComponent(OriginDetailComponent);
+		component = fixture.componentInstance;
+		fixture.detectChanges();
+		await fixture.whenStable();
+		expect(paramMap).toHaveBeenCalled();
+		expect(component.origin).not.toBeNull();
+		expect(component.origin.name).toBe(origin.name);
+		expect(component.new).toBeFalse();
+	});
+
+	it("deletes existing Origins", async () => {
+		const spy = spyOn(service, "deleteOrigin").and.callThrough();
+		let cgs = await service.getOrigins();
+		const initialLength = cgs.length;
+		if (initialLength < 1) {
+			return fail("need at least one Cache Group");
+		}
+		const cg = cgs[0];
+		component.origin = cg;
+		component.new = false;
+
+		const asyncExpectation = expectAsync(component.deleteOrigin()).toBeResolvedTo(undefined);
+		const dialogs = await loader.getAllHarnesses(MatDialogHarness);
+		if (dialogs.length !== 1) {
+			return fail(`failed to open dialog; ${dialogs.length} dialogs found`);
+		}
+		const dialog = dialogs[0];
+		const buttons = await dialog.getAllHarnesses(MatButtonHarness.with({text: /^[cC][oO][nN][fF][iI][rR][mM]$/}));
+		if (buttons.length !== 1) {
+			return fail(`'confirm' button not found; ${buttons.length} buttons found`);
+		}
+		await buttons[0].click();
+
+		expect(spy).toHaveBeenCalledOnceWith(cg);
```

Comment:
You're explicitly calling `deleteOrigin` with the **ID** of the Origin being deleted, not the entire Origin. So once you fix the testing setup, this will fail, because it'll expect a call with a whole origin passed but you're passing a number.